src/Microsoft.ML.Core/Utilities/OrderedWaiter.cs

```diff
@@ -90,7 +90,7 @@ public void Wait(long position, CancellationToken token = default(CancellationTo
                 ev = new WaitStats(position);
                 _waiters.Add(ev);
             }
-            ev.Event.Wait(token);
+            ev.Event.Wait(5*60*1000, token);
             if (_ex != null)
```

Comment:
Can this be a constant that will apply to all use cases? Or does this need to be a parameter to the function? If it is a constant, can you please give it a name and use that instead?

Reply:
Yes, this should, this number is not reliable now so just put the number directly, let's discuss more in person.

---
In reply to: [350531730](https://github.com/dotnet/machinelearning/pull/4503#discussion_r350531730) [](ancestors = 350531730)